Zend/tests/gh7900.phpt

```diff
@@ -0,0 +1,23 @@
+--TEST--
+GH-7900: Arrow function with never return type compile-time errors
+--FILE--
+<?php
+
+$x = fn(): never => throw new \Exception('Here');
+
+try {
+    var_dump($x());
+} catch (\Exception $e) {
+    echo $e->getMessage(), "\n";
+}
+
+try {
+    assert((fn(): never => 42) && false);
```

Comment:
Is `assert()` guaranteed to throw in PHPTs? Otherwise there would be the need to explicitly set some INI values.